### PR TITLE
[JVM_IR] Give temporary variables meaninful offsets.

### PR DESCRIPTION
**Pull request by madsager**
Temporary variable loads in when expressions had the offsets
of the variable declaration. That leads to hacks during codegen
for line number generation.

Instead of those hacks, give the variable loads the offsets of
the context in which they occur. That avoids the codegen hacks
and fixes stepping behavior for more when expressions.

### CODE REVIEW

**Comment by madsager** [comment]
First commit is https://github.com/JetBrains/kotlin/pull/3633, fixes in second commit.

**Comment by madsager** [comment]
Added a commit to allow ir specific expectations for the KotlinSteppingTests and used that to have reasonable expectations for a constructor test that has reasonable behavior on the JVM_IR backend.

File: idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/IrKotlinSteppingTestGenerated.java [hunk 1]
```diff
@@ -0,0 +1,1310 @@
...
+        }
+
+        @TestMetadata("smartStepIntoInlinedFunLiteral.kt")
+        public void testSmartStepIntoInlinedFunLiteral() throws Exception {
+            runTest("idea/jvm-debugger/jvm-debugger-test/testData/stepping/custom/smartStepIntoInlinedFunLiteral.kt");
+        }
+
+        @TestMetadata("smartStepIntoInlinedFunctionalExpression.kt")
+        public void testSmartStepIntoInlinedFunctionalExpression() throws Exception {
+            runTest("idea/jvm-debugger/jvm-debugger-test/testDat" +
```
Comment (max-kammerer) [inline idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/IrKotlinSteppingTestGenerated.java:1241]: Please regenerate file

Comment (max-kammerer) [reply idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/IrKotlinSteppingTestGenerated.java:1241]: Seem that changes in Fir** tests already  pushed to master

Comment (madsager) [reply idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/IrKotlinSteppingTestGenerated.java]: Whoops, yes, rebased.

**Comment by max-kammerer** [comment]
@madsager Please rebase on master

**Comment by madsager** [comment]
> @madsager Please rebase on master

Done.

**Comment by max-kammerer** [comment]
Thank you!

**Pull request merged**